### PR TITLE
B2CA-1272: Remove throw from App

### PR DESCRIPTION
**Pull request by cedelavergne-ledger**
Cleanup the full App:

- Use full standard app (except `main.c`, which is dedicated because too different from SDK)
- Cleanup `APDU_RESPONSE_xx` code and remove hardcoded values in the source code
- Replace the `THROW` by return codes
- Replace most `LEDGER_ASSERT` and `CX_ASSERT` by return codes
- Improve and factorize `io_seproxyhal_xxx` functions
- Fix duplicated header filenames between sdk and app
- Fix #B2CA-1583 (App blocked after unplug usb when rejecting a transaction)
- Cleanup main.c

## Changes include

- [ ] Bugfix (non-breaking change that solves an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (change that is not backwards-compatible and/or changes current functionality)
- [ ] Tests
- [X] Documentation
- [X] Other (for changes that might not fit in any category)

### CODE REVIEW

File: src_features/signTx/logic_signTx.c [hunk 1]
```diff
@@ -360,7 +372,7 @@ __attribute__((noinline)) static bool finalize_parsing_helper(void) {
         if (!eth_plugin_call(ETH_PLUGIN_FINALIZE, (void *) &pluginFinalize)) {
             PRINTF("Plugin finalize call failed\n");
             report_finalize_error();
-            return false;
+            return 0;
```
Comment (apaillier-ledger) [inline src_features/signTx/logic_signTx.c:375]: Not a valid SW

**Review by apaillier-ledger — CHANGES_REQUESTED** [review]

File: src/ethUstream.c [hunk 2]
```diff
@@ -21,11 +21,43 @@
 #include "ethUstream.h"
 #include "rlp_utils.h"
 #include "common_utils.h"
+#include "ledger_assert.h"
```
[the remark this answers is not in the document]
Comment (cedelavergne-ledger) [reply src/ethUstream.c:24]: Correct. Fixed!

File: src/main.c [hunk 3]
```diff
@@ -134,253 +107,141 @@ const uint8_t *parseBip32(const uint8_t *dataBuffer, uint8_t *dataLength, bip32_
...
 }
 
 void app_main(void) {
-    unsigned int rx = 0;
-    unsigned int tx = 0;
-    unsigned int flags = 0;
+    uint32_t rx = 0;
+    uint32_t tx = 0;
+    uint32_t flags = 0;
+    uint32_t sw = 0;
```
[the remark this answers is not in the document]
Comment (cedelavergne-ledger) [reply src/main.c:242]: Fixed

File: src/apdu_constants.h [hunk 4]
```diff
@@ -33,100 +34,72 @@
...
 #define APDU_RESPONSE_REF_DATA_NOT_FOUND      0x6a88
+#define APDU_RESPONSE_INVALID_P1_P2           0x6b00
+#define APDU_RESPONSE_INVALID_INS             0x6d00
+#define APDU_RESPONSE_INVALID_CLA             0x6e00
 #define APDU_RESPONSE_UNKNOWN                 0x6f00
+#define APDU_RESPONSE_OK                      0x9000
+#define APDU_RESPONSE_CMD_CODE_NOT_SUPPORTED  0x911c
 
-enum { OFFSET_CLA = 0, OFFSET_INS, OFFSET_P1, OFFSET_P2, OFFSET_LC, OFFSET_CDATA };
+uint32_t handleGetPublicKey(uint8_t p1,
```
[the remark this answers is not in the document]
Comment (cedelavergne-ledger) [reply src/apdu_constants.h:56]: Fixed

File: src/main.c [hunk 5]
```diff
@@ -134,253 +107,141 @@ const uint8_t *parseBip32(const uint8_t *dataBuffer, uint8_t *dataLength, bip32_
...
+    dataBuffer += bip32->length * sizeof(uint32_t);
+    *dataLength -= bip32->length * sizeof(uint32_t);
 
     return dataBuffer;
 }
 
-void handleApdu(unsigned int *flags, unsigned int *tx) {
-    unsigned short sw = 0;
+static uint32_t handleApdu(command_t *cmd, uint32_t *flags, uint32_t *tx) {
+    uint32_t sw = 0;
```
[the remark this answers is not in the document]
Comment (cedelavergne-ledger) [reply src/main.c:122]: Fixed

File: src/eth_swap_utils.h [hunk 6]
```diff
@@ -18,9 +18,24 @@
 #pragma once
 
 #include <stdint.h>
+#include "swap_lib_calls.h"
+#include "chainConfig.h"
+#include "caller_api.h"
 
-bool parse_swap_config(const uint8_t* config,
+typedef struct eth_libargs_s {
```
Comment (apaillier-ledger) [inline src/eth_swap_utils.h:25]: Why is the definition of this struct duplicated from eth_swap_lib_calls.h ?

**Review by apaillier-ledger — CHANGES_REQUESTED** [review]

File: src/eth_swap_utils.h (hunk 6 above)
Comment (cedelavergne-ledger) [reply src/eth_swap_utils.h:25]: Good catch. Copy-Past error :wink:

**Review by apaillier-ledger — APPROVED** [review]

**Pull request merged**